### PR TITLE
Add Crayons classes to the "Destroy your account" page

### LINKED ISSUE #12296: Add Crayons to the "Destroy your account" page
**Describe the bug**

Right now the "Destroy your account" page looks like this:

Screenshot_2021-01-15 Screenshot

**To Reproduce**

1. Turn on caching with `bundle exec rails dev:cache`
2. Go to http://localhost:3000/settings/account
3. Click on "Delete account"
4. Check the logs and copy the URL, it should be something like `http://localhost:3000/users/confirm_destroy/$DESTROY_TOKEN` (the token lasts 12 hours)
5. Go to said URL and see the page

**Expected behavior**

Something more pleasing to the eyes :D

### PR DESCRIPTION
**Pull request by narender2031**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
 refactor the UI for the delete my account page
## Related Tickets & Documents
Closes #12296

## QA Instructions, Screenshots, Recordings

Screenshot 2021-01-20 at 11 35 18 AM

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [ ] Yes
- [x] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

### CODE REVIEW

**Comment by rhymes** [comment]
@narender2031 I converted the branch to draft 👍

**Comment by narender2031** [comment]
@ludwiczakpawel any update on this PR?

File: app/views/users/confirm_destroy.html.erb [hunk 1]
```diff
@@ -1,55 +1,59 @@
...
-    </script>
-    <p>
-      If you would like to keep your content under the <%= link_to "@ghost", "/ghost" %> account,
-      please <%= email_link(text: "click here", additional_info: { subject: "Request Account Deletion", body: @email_body }) %>
-    </p>
+<div class="user-settings-page crayons-layout crayons-layout--limited-l gap-0">
+  <div class="crayons-card min-w-0 text-padding">
+    <h2 class="crayons-title"> Destroy your account</h2>
+    <div class="settings-form">
+      <h4 class="fw-normal">
```
Comment (ludwiczakpawel) [inline app/views/users/confirm_destroy.html.erb:5]: Let's drop this entire `h4` - if anything, it should be `div` but in this case i don't think we need another container here.

```suggestion
```

File: app/views/users/confirm_destroy.html.erb [hunk 2]
```diff
@@ -1,55 +1,59 @@
...
+        <%= form_tag user_full_delete_path, method: :delete, autocomplete: "off", id: "delete__account" do %>
+          Deleting your account will:
+          <ul class="delete__account mb-4">
```
Comment (ludwiczakpawel) [inline app/views/users/confirm_destroy.html.erb:8]:
```suggestion
          <ul class="my-0 ml-6 list-disc">
```

^ dropping unnecessary class name and adding a bit of styling.

File: app/views/users/confirm_destroy.html.erb [hunk 3]
```diff
@@ -1,55 +1,59 @@
...
+            </label>
+            <input type="text" name="delete__account__username__field" class="crayons-textfield " id="delete__account__username__field" placeholder="username">
+          </div>
+          <div class="field mb-4 w-50">
+            <label for="delete__account__verification__field" class="crayons-field__label">
+              For verification, type the words:
+              <p class="crayons-field__description"> delete my account</p>
+            </label>
+            <input type="text" name="delete__account__verification__field" class="crayons-textfield" id="delete__account__verification__field">
+          </div>
```
Comment (ludwiczakpawel) [inline app/views/users/confirm_destroy.html.erb:31]:
```suggestion
        <div class="crayons-field m:w-50">
          <label for="delete__account__username__field" class="crayons-field__label">
            Type in your username
          </label>
          <input type="text" name="delete__account__username__field" class="crayons-textfield " id="delete__account__username__field" placeholder="username">
        </div>
        <div class="crayons-field m:w-50">
          <label for="delete__account__verification__field" class="crayons-field__label">
            For confirmaiton, type the words: <code>delete my account</code>
          </label>
          <input type="text" name="delete__account__verification__field" class="crayons-textfield" id="delete__account__verification__field">
        </div>
```

- this is how `.crayons-field` component should be used with pure HTML so it does render properly
- I've also suggested some changes to the copy as well.

File: app/views/users/confirm_destroy.html.erb [hunk 4]
```diff
@@ -1,55 +1,59 @@
...
-    <p>
-      If you would like to keep your content under the <%= link_to "@ghost", "/ghost" %> account,
-      please <%= email_link(text: "click here", additional_info: { subject: "Request Account Deletion", body: @email_body }) %>
-    </p>
+<div class="user-settings-page crayons-layout crayons-layout--limited-l gap-0">
+  <div class="crayons-card min-w-0 text-padding">
+    <h2 class="crayons-title"> Destroy your account</h2>
+    <div class="settings-form">
+      <h4 class="fw-normal">
+        <%= form_tag user_full_delete_path, method: :delete, autocomplete: "off", id: "delete__account" do %>
```
Comment (ludwiczakpawel) [inline app/views/users/confirm_destroy.html.erb:6]:
```suggestion
        <%= form_tag user_full_delete_path, method: :delete, autocomplete: "off", id: "delete__account", class: 'flex flex-col gap-4 mb-4' do %>
```

`flex flex-col gap-4` will automatically apply spacing between its children elements.

File: app/views/users/confirm_destroy.html.erb [hunk 5]
```diff
@@ -1,55 +1,59 @@
...
+          <ul class="delete__account mb-4">
+            <li>delete your profile, along with your Twitter and/or GitHub associations.
+              This does not include app permissions. You will have to remove them yourself on
+              <a href="https://twitter.com/settings/applications">Twitter profile settings</a> and
+              <a href="https://github.com/settings/applications">GitHub profile settings</a>.
+            </li>
+            <%# TODO: expand the delete messaging later %>
+            <li>delete any and all content you have, such as articles, comments, your reading list or chat messages.</li>
+            <li>allow your username to become available to anyone.</li>
+          </ul>
```
Comment (ludwiczakpawel) [inline app/views/users/confirm_destroy.html.erb:17]:
```suggestion
          </ul>
          <h3>To delete your account:</h3>
```

This should structure everything slightly better.

File: app/views/users/confirm_destroy.html.erb [hunk 6]
```diff
@@ -1,55 +1,59 @@
...
+              <p class="crayons-field__description"> delete my account</p>
+            </label>
+            <input type="text" name="delete__account__verification__field" class="crayons-textfield" id="delete__account__verification__field">
+          </div>
+          <button class="crayons-btn crayons-btn--l crayons-btn--danger" id="delete__account__btn" type="submit" disabled="true">
+            DELETE ACCOUNT
+          </button>
+        <% end %>
+      </h4>
+      <br>
```
Comment (ludwiczakpawel) [inline app/views/users/confirm_destroy.html.erb:37]:
```suggestion
```

File: app/views/users/confirm_destroy.html.erb [hunk 7]
```diff
@@ -1,55 +1,59 @@
...
-      deleteAccountUsernameInput.addEventListener('input', disableDeleteAccountBtn);
-      deleteAcccountVerificationInput.addEventListener('input', disableDeleteAccountBtn);
-    </script>
-    <p>
-      If you would like to keep your content under the <%= link_to "@ghost", "/ghost" %> account,
-      please <%= email_link(text: "click here", additional_info: { subject: "Request Account Deletion", body: @email_body }) %>
-    </p>
+<div class="user-settings-page crayons-layout crayons-layout--limited-l gap-0">
+  <div class="crayons-card min-w-0 text-padding">
+    <h2 class="crayons-title"> Destroy your account</h2>
```
Comment (ludwiczakpawel) [inline app/views/users/confirm_destroy.html.erb:3]:
```suggestion
    <h2 class="crayons-title mb-4"> Destroy your account</h2>
```

^ adding a bit of spacing below page title.

**Review by ludwiczakpawel — CHANGES_REQUESTED** [review]
It's definitely a good start although I've suggested several code changes to make it look more readable and structured properly.

**Comment by narender2031** [comment]
@ludwiczakpawel can you please re-start the travis build again

**Comment by ludwiczakpawel** [comment]
Oh it's actually properly failed. This `click_button` in spec is case-sensitive, so you may need to update this to "Delete account": https://github.com/forem/forem/blob/master/spec/system/user/user_self_destroy_spec.rb#L50

**Review by ludwiczakpawel — APPROVED** [review]
👍

**Review by rhymes — APPROVED** [review]

**Pull request merged**